lighthouse-core/audits/byte-efficiency/cache-headers.js

```diff
@@ -0,0 +1,225 @@
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const assert = require('assert');
+const parseCacheControl = require('parse-cache-control');
+const ByteEfficiencyAudit = require('./byte-efficiency-audit');
+const formatDuration = require('../../report/v2/renderer/util.js').formatDuration;
+const WebInspector = require('../../lib/web-inspector');
+const URL = require('../../lib/url-shim');
+
+// Ignore assets that have very high likelihood of cache hit
+const IGNORE_THRESHOLD_IN_PERCENT = 0.9;
+// Basically we assume a 10% chance of repeat visit.
+const PROBABILITY_OF_RETURN_VISIT = 0.1;
+
+class CacheHeaders extends ByteEfficiencyAudit {
+  /**
+   * @return {number}
+   */
+  static get PROBABILITY_OF_RETURN_VISIT() {
+    return PROBABILITY_OF_RETURN_VISIT;
+  }
+
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'Caching',
+      name: 'cache-headers',
+      informative: true,
+      helpText:
+        'A well-defined cache policy can speed up repeat visits to your page. ' +
+        '[Learn more](https://developers.google.com/speed/docs/insights/LeverageBrowserCaching).',
+      description: 'Leverage browser caching for static assets',
+      requiredArtifacts: ['devtoolsLogs'],
+    };
+  }
+
+  /**
+   * Interpolates the y value at a point x on the line defined by (x0, y0) and (x1, y1)
+   * @param {number} x0
+   * @param {number} y0
+   * @param {number} x1
+   * @param {number} y1
+   * @param {number} x
+   * @return {number}
+   */
+  static linearInterpolation(x0, y0, x1, y1, x) {
+    const slope = (y1 - y0) / (x1 - x0);
+    return y0 + (x - x0) * slope;
+  }
+
+  /**
+   * Computes the percent likelihood that a return visit will be within the cache lifetime, based on
+   * Chrome UMA stats see the note below.
+   * @param {number} maxAgeInSeconds
+   * @return {number}
+   */
+  static getCacheHitProbability(maxAgeInSeconds) {
+    // This array contains the hand wavy distribution of the age of a resource in hours at the time of
+    // cache hit at 0th, 10th, 20th, 30th, etc percentiles. This is used to compute `wastedMs` since there
+    // are clearly diminishing returns to cache duration i.e. 6 months is not 2x better than 3 months.
+    // Based on UMA stats for HttpCache.StaleEntry.Validated.Age, see https://www.desmos.com/calculator/7v0qh1nzvh
+    // Example: a max-age of 12 hours already covers ~50% of cases, doubling to 24 hours covers ~10% more.
+    const RESOURCE_AGE_IN_HOURS_DECILES = [0, 0.2, 1, 3, 8, 12, 24, 48, 72, 168, 8760, Infinity];
+    assert.ok(RESOURCE_AGE_IN_HOURS_DECILES.length === 12, '1 for each decile, 1 on each boundary');
```

Comment:
> Looks like Infinity might be for a past the upper bound check (boundary somehow past the 100th percentile 

Yeah because 1 year+ is all basically 100th but doing linear interpolation with Infinity isn't quite fair :) I guess it doesn't really matter that much though if we ignore 90% and up, how about I replace with `Number.MAX_VALUE` and nuke the check?